pyoptsparse/pyALPSO/pyALPSO.py

```diff
@@ -226,7 +226,7 @@ def _on_setOption(self, name, value):
             else:
                 raise ValueError("parallel_type must be either '' or 'EXT'.")
 
-    def _on_getOption(self, name, value):
```

Comment:
We did not caught this before because the `_on_getOption` methods are mostly used (except for SNOPT, IPOPT, ...) to overwrite the same method from parent `Optimimzer` class right?
Is there a more elegant way to check for the existence of this method in the single optimizer wrappers? Does using a try/except approach become more cumbersome?

Reply:
This was always a bug, but was never caught because previously the `getOption` function defined in the Optimizer class would first `return` then execute `_on_getOption`, so obviously the optimizer-specific `_on_getOption` was never executed. I have swapped the order so that the return statement is at the end of the function.

Reply:
Oh gotcha, so basically we are just picking from the dict we set in `setOption()`. I just checked and the `_on_getOption` is a `pass` function in every optimizer wrapper we support now, which further explains why we never bothered to catch this. I will look at the refactoring you made but I feel it makes much more sense to just have everything passing in the parent `Optimizer` class and in case add more, rather than having to override the error in every wrapper

Reply:
Yep agreed, I've updated this now.